nodeadm/internal/api/types.go

```diff
@@ -43,15 +39,6 @@ type InstanceDetails struct {
 	MAC              string `json:"mac,omitempty"`
 }
 
-type ClusterDetails struct {
-	Name                 string `json:"name,omitempty"`
-	APIServerEndpoint    string `json:"apiServerEndpoint,omitempty"`
-	CertificateAuthority []byte `json:"certificateAuthority,omitempty"`
-	CIDR                 string `json:"cidr,omitempty"`
-	EnableOutpost        *bool  `json:"enableOutpost,omitempty"`
-	ID                   string `json:"id,omitempty"`
-}
-
```

Comment:
Ah, you'll have to add this back, the internal API should be separate from the public API types. The internal/api/bridge package maps multiple supported versions of the public API onto a single internal representation, that should be used everywhere. That package is the only place under internal/ where the public API types should be imported

Reply:
oh gotcha, i saw other projects using public types in their impls and misinterpreted this, i can quickly flip them all back